integration/simulation/validate_chain.go

```diff
@@ -363,3 +365,40 @@ func extractWithdrawals(t *testing.T, nodeClient rpc.Client, nodeAddr uint64) (t
 		}
 	}
 }
+
+// Terminates all subscriptions, and checks that we received events for both the HOC and POC ERC20 contracts.
+// TODO - #453 - Extend logic of this test.
+func checkLogsReceived(t *testing.T, s *Simulation) {
+	// In-memory clients cannot handle subscriptions for now.
+	if s.Params.IsInMem {
+		return
+	}
+
+	for _, sub := range s.Subscriptions {
+		sub.Unsubscribe()
+	}
+
+	var gotHOCEvent bool
+	var gotPOCEvent bool
+	for {
+		select {
+		case receivedLog := <-s.LogChannel:
+			if receivedLog.Address.Hex() == "0x"+bridge.HOCAddr {
```

Comment:
is it possible to check other fields of the log entry as well?

Reply:
@tudor-malene Yeah, I can add that in a follow-up. What else would you like me to check?

Reply:
like the topic. donno.